service/controller/v2/resource/deployment/resource.go

```diff
@@ -172,6 +196,20 @@ func getDeploymentByName(list []deployment, name string) (deployment, error) {
 	return deployment{}, microerror.Maskf(notFoundError, name)
 }
 
+func isFinalState(s string) bool {
+	if s == "Succeeded" {
+		return true
+	}
+	if s == "Failed" {
+		return true
+	}
+	if s == "Canceled" {
```

Comment:
`Failed` and `Canceled` should be treated as errors IMO

Reply:
What should then happen?

Reply:
Failed probably means the deployment failed. I think we should log the state in general. So we can make other decisions in the future on how to deal with that.

Reply:
Any cool way of logging the state? What is important there? Feels like premature optimization. We guess into the blue and have no use case. I checked the [`DeploymentExtended`](https://godoc.org/github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-02-01/resources#DeploymentExtended). It has only pointers and such so we would need to marshal it and shit.

Reply:
We used to have state from `_, err := DeploymentsCreateOrUpdateFuture.Done()` which was quite useful to quickly see what went wrong from the logs.

Logging state from `DeploymentExtended` seems more complex, so what about `Name`, `ID`, `Properties. CorrelationID` and `Properties.Timestamp` for now. So we can at least look up from Azure Portal what is going wrong in case of failure.

Reply:
Is it ok to track better debugging in a separate issue?

Reply:
ok

Reply:
See https://github.com/giantswarm/giantswarm/issues/3290.

Reply:
I think it's good to see in the logs that deployment failed. That can be to various reasons, e.g. be invalid json or non-existing resource id, or infinite number of other reasons. If that shows up as an error, fine. But I'm not sure about it.